### PR TITLE
Check eligibility status after enabling the share all current and future addresses toggle

### LINKED ISSUE #18867: Eligibility status is not updated after selecting all account by toggle
cc @ajayesivan 

**Steps:** 

1. Open community that has set permissions
2. Go to 'Addresses for permissions' screen
3. Disable 'Select all' toggle
4. Unselect some account that affects eligibility status 
5. Wait for eligibility status to be updated
6. Enable select all toggle
7. See if eligibility status has changed to correct one

**Actual result:** eligibility status does not update. It remains the same as was after unselecting account. 

https://github.com/status-im/status-mobile/assets/97245802/02f10062-b20e-4ce9-93cf-d54d09df948b

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by ajayesivan**
fixes #18867 

status: ready

### CODE REVIEW

**Review by cammellos — APPROVED** [review]

**Review by ilmotta — APPROVED** [review]

**Review by ibrkhalil — APPROVED** [review]

**Review by siddarthkay — APPROVED** [review]

File: src/status_im/contexts/communities/events.cljs [hunk 1]
```diff
@@ -223,16 +223,19 @@
...
     {:db (update-in db
                     [:communities community-id]
-                    (fn [community]
-                      (-> community
-                          (assoc :share-all-addresses? (not share-all-addresses?))
-                          (cond-> (not share-all-addresses?)
-                                  (assoc :selected-permission-addresses addresses)))))}))
+                    assoc
+                    :share-all-addresses?          next-share-all-addresses?
+                    :selected-permission-addresses addresses)
```
Comment (Parveshdhull) [inline src/status_im/contexts/communities/events.cljs:234]: Hi @ajayesivan,
wouldn't this always share all addresses when we are toggling? (even if we are disabling sharing)

Comment (Parveshdhull) [reply src/status_im/contexts/communities/events.cljs:234]: ok got, this is expected behavior. We keep shared all addresses, but only change if we are allowed to deselect or not, based on `share-all-addresses?` flag

**Review by Parveshdhull — APPROVED** [review]

File: src/status_im/contexts/communities/events.cljs (hunk 1 above)
Comment (ajayesivan) [reply src/status_im/contexts/communities/events.cljs:234]: Disabling the toggle implies that the previous state was enabled, indicating that all addresses were selected. Consequently, in both situations, whether the toggle is enabled or disabled, the sharing of all addresses doesn't make any difference. Do you see any potential issues with this, or does it make sense to you?

**Comment by qoqobolo** [comment]
Hey @ajayesivan, thanks for the fix, works as expected for me.
This PR can be merged. 
cc @cammellos

**Pull request merged**